src/Authentication/Authentication/Cmdlets/InvokeGraphRequest.cs

```diff
@@ -0,0 +1,965 @@
+                }
+            }
+            catch (HttpRequestException httpRequestException)
+            {
+                WriteError(new ErrorRecord(httpRequestException, ErrorCategory.ConnectionError.ToString(),
+                    ErrorCategory.InvalidResult, null));
+                throw;
+            }
+            catch (Exception exception)
+            {
+                WriteError(new ErrorRecord(exception, ErrorCategory.ConnectionError.ToString(),
+                    ErrorCategory.InvalidOperation, null));
+                throw;
+            }
+        }
+
+        protected override void EndProcessing()
+        {
+            base.EndProcessing();
+        }
+
+        protected override void StopProcessing()
+        {
+            base.StopProcessing();
+        }
+        #endregion
+        private static ErrorRecord GenerateHttpErrorRecord(HttpMessageFormatter httpResponseMessageFormatter, HttpRequestMessage httpRequestMessage)
+        {
+            var currentResponse = httpResponseMessageFormatter.HttpResponseMessage;
+            var errorMessage = StringFormatCurrentCulture("ResponseStatusCodeFailure {0} {1}", currentResponse.StatusCode, currentResponse.ReasonPhrase);
+            var httpException = new HttpResponseException(errorMessage, currentResponse);
+            var errorRecord = new ErrorRecord(httpException, "WebCmdletWebResponseException", ErrorCategory.InvalidOperation, httpRequestMessage);
+            var detailMsg = httpResponseMessageFormatter.ReadAsStringAsync()
+                .GetAwaiter()
+                .GetResult();
+            if (!string.IsNullOrEmpty(detailMsg))
+            {
+                errorRecord.ErrorDetails = new ErrorDetails(detailMsg);
+            }
+
+            return errorRecord;
+        }
+
+        private void ReportRequestStatus(HttpRequestMessage requestMessage)
+        {
+            long requestContentLength = 0;
+            if (requestMessage.Content != null)
+            {
+                requestContentLength = requestMessage.Content.Headers.ContentLength.Value;
+            }
+            var reqVerboseMsg = StringFormatCurrentCulture("{0} {1} with {2}-byte payload",
+                requestMessage.Method,
+                requestMessage.RequestUri,
+                requestContentLength);
+            WriteVerbose(reqVerboseMsg);
+        }
+
+        private void ReportResponseStatus(HttpResponseMessage responseMessage)
+        {
+            var contentType = ContentHelper.GetContentType(responseMessage);
+            var respVerboseMsg = StringFormatCurrentCulture("received {0}-byte response of content type {1}",
+                responseMessage.Content.Headers.ContentLength,
+                contentType);
+            WriteVerbose(respVerboseMsg);
+        }
+
+        private static string StringFormatCurrentCulture(string format, params object[] args)
+        {
+            return string.Format(CultureInfo.CurrentCulture, format, args);
+        }
```

Comment:
can be replaced with the function you are defining in `StringUtil`.

Reply:
Moved to StringUtil